### PR TITLE
@W-7978510@: Initial partial implementation of RetireJS engine. Implementation of scanner:run command still lacking.

### PR DESCRIPTION
**Pull request by jfeingold35**

### CODE REVIEW

File: src/lib/util/Config.ts [hunk 1]
```diff
@@ -54,7 +54,16 @@ const DEFAULT_CONFIG: ConfigContent = {
...
-        }
+        },
+		{
+			name: ENGINE.RETIRE_JS,
+			targetPatterns: [
+				'**/*.js',
+				'!**/node_modules/**',
+				'!**/bower_components/**'
+			],
+			disabled: false
```
Comment (jfeingold35) [inline src/lib/util/Config.ts:65]: All `2.x` releases are going to be based off the `release` branch, right? If that's the case, then I think we can leave this engine enabled by default.

Comment (jbartolotta-sfdc) [reply src/lib/util/Config.ts:65]: 2.x will be based off of `dev`. I think it would be best to disable this by default.

Comment (jfeingold35) [reply src/lib/util/Config.ts:65]: Cool. Then we should also wait to merge this until we fix the bug that includes disabled engines in the output of `scanner:rule:list` and `scanner:rule:describe`, right?

**Review by jbartolotta-sfdc — COMMENTED** [review]
This looks good, we plan do release from the `dev` branch next week. Do you want to create a new retire-js branch to merge these changes into? Releasing this without the backing implementation next week could confuse users.

**Comment by jfeingold35** [comment]
> This looks good, we plan do release from the `dev` branch next week. Do you want to create a new retire-js branch to merge these changes into? Releasing this without the backing implementation next week could confuse users.

Either that or we just keep using this feature branch until the implementation is complete. Your way is probably better for branch hygiene, though. I'll fork a branch off of Dev shortly.

File: src/lib/retire-js/RetireJsEngine.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+//import path = require('path');
+
+// Unlike the other engines we use, RetireJS doesn't really have "rules" per se. So we sorta have to synthesize a
+// "catalog" out of RetireJS's normal behavior.
+const retireJsCatalog: Catalog = {
+	rules: [{
+		engine: ENGINE.RETIRE_JS.valueOf(),
+		sourcepackage: ENGINE.RETIRE_JS.valueOf(),
+		// Give this rule an informative name, specific enough that we're able to supplement it with other rules later.
+		name: ' insecure-bundled-dependencies',
```
Comment (rmohan20) [inline src/lib/retire-js/RetireJsEngine.ts:17]: Is the `[space]` at the beginning of the name intentional?

File: src/lib/retire-js/RetireJsEngine.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	}],
+	rulesets: []
+};
+
+export class RetireJsEngine implements RuleEngine {
+	public static ENGINE_ENUM: ENGINE = ENGINE.RETIRE_JS;
+	public static ENGINE_NAME: string = ENGINE.RETIRE_JS.valueOf();
+	// RetireJS isn't really built to be invoked programmatically, so we'll need to invoke it as a CLI command. However, we
+	// can't assume that they have the module installed globally. So what we're doing here is identifying the path to the
+	// locally-scoped `retire` module, and then using that to derive a path to the CLI-executable JS script.
```
Comment (rmohan20) [inline src/lib/retire-js/RetireJsEngine.ts:37]: Am I correct in assuming that the locally-scoped `retire` module will be from within our package? Also, I can't find this logic in the code - can I assume that this logic will be included as a part of `scanner:run` changes?

**Review by rmohan20 — APPROVED** [review]
Just a couple of minor questions. Looks good 👍

File: src/lib/retire-js/RetireJsEngine.ts (hunk 2 above)
Comment (jfeingold35) [reply src/lib/retire-js/RetireJsEngine.ts:17]: No, it's a mistake. I've fixed it in the branch that implements the`run` command though, so we don't need to fix it here.

File: src/lib/retire-js/RetireJsEngine.ts (hunk 3 above)
Comment (jfeingold35) [reply src/lib/retire-js/RetireJsEngine.ts:37]: Correct. The logic at line 38 is identifying the path to the local RetireJS module. It's commented out so the tests pass, and actually implemented in the `run` PR.

**Pull request merged**